### PR TITLE
Hashes and dashes

### PR DESCRIPTION
**Pull request by filiptronicek**
🔒

## 📝 Release notes 
**BREAKING**: clip codes are now 99 characters long, and they can be accessed by any 5+ character-long substring of this code. Codes are still base 36, so there's an equal number of permutations for the 5-character combination and a nearly infinite space to scale to.

### CODE REVIEW

File: src/lib/generateID.ts [hunk 1]
```diff
@@ -1,18 +1,11 @@
...
-      (x, i) => (
-        (i = ((x / 255) * 61) | 0),
-        String.fromCharCode(i + (i > 9 ? (i > 35 ? 61 : 55) : 48))
-      ),
-    )
-    .join('')
-    .toLowerCase();
+export const getClipHash = (url: string): string => {
+  const digest = crypto.createHash('sha512').update(url).digest('hex');
+  return BigInt('0x' + digest).toString(36);
```
Comment (bigint) [inline src/lib/generateID.ts:10]: bigint ❤️

Comment (filiptronicek) [reply src/lib/generateID.ts:10]: That's exactly what popped into my head 😄

Comment (bigint) [reply src/lib/generateID.ts:10]: Ethereum uses bigint 😄

Comment (filiptronicek) [reply src/lib/generateID.ts:10]: Welp Interclip now does too!

Comment (bigint) [reply src/lib/generateID.ts:10]: Then let us build Interclip blockchain 🤣

File: src/pages/api/clip/set.ts [hunk 2]
```diff
@@ -58,19 +56,49 @@ export default async function handler(
...
+      data: {
+        hashLength: equal + 1,
+        code: newHash,
+      },
+    });
+
+    res.status(200).json({
+      status: 'success',
+      result: { ...existingClip, hashLength: equal + 1 },
+    });
```
Comment (bigint) [inline src/pages/api/clip/set.ts:96]: Please don't do DB collision checks 😭

**Review by bigint — APPROVED** [review]
LGTM 🚀

**Pull request merged**